### PR TITLE
feat(student-dashboard): submit tasks to random review

### PR DESCRIPTION
**Pull request by nataliagulko**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [ ] I followed naming convention rules

---

#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [x] Refactoring
- [x] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

#1851, #1586

#### 💡 Background and solution

1. Student Dashboard:
    1. Split MentorCard to several components
    2. Extract useReducer logic from MentorCard to custom hook
    3. Isolate modal logic from MentorCard logic
    4. Display "Submit task" button when student doesn't have a mentor
    5. Update MentorCard layout
    6. Add tests
1. Mentor Dashboard:
    1. Display "Random task" tab
    2. Get task for review from students which don't have mentors

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: client/src/modules/StudentDashboard/components/MentorCard/MentorInfo/MentorInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { Col, Row, Space, Typography } from 'antd';
```
[the remark this answers is not in the document]
Comment (apalchys) [reply client/src/modules/StudentDashboard/components/MentorCard/MentorInfo/MentorInfo.tsx:1]: and the same with other components there

File: client/src/modules/StudentDashboard/hooks/useSubmitTaskSolution.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const [state, dispatch] = useReducer(reducer, null);
+
+  const showModal = async () => {
+    dispatch({ type: 'loading' });
+    const coursesTasksApi = new CoursesTasksApi();
+    const { data } = await coursesTasksApi.getCourseTasks(courseId);
+    const courseTasks = data.filter(
+      item =>
+        item.checker === CreateCourseTaskDtoCheckerEnum.Mentor &&
+        item.type != 'selfeducation' &&
```
Comment (apalchys) [inline client/src/modules/StudentDashboard/hooks/useSubmitTaskSolution.ts:45]: please check if we have type enum in `api`. If so, I suggest to use it

File: client/src/modules/StudentDashboard/components/MentorCard/MentorInfo/MentorInfo.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nataliagulko) [reply client/src/modules/StudentDashboard/components/MentorCard/MentorInfo/MentorInfo.tsx]: Fixed, thank you

File: client/src/modules/StudentDashboard/hooks/useSubmitTaskSolution.ts (hunk 2 above)
Comment (nataliagulko) [reply client/src/modules/StudentDashboard/hooks/useSubmitTaskSolution.ts]: Fixed, thank you

**Pull request merged**